### DISCUSSION TITLE
Question about SRP server published services

### DISCUSSION THREAD

**Discussion by suveshpratapa** [openthread/openthread#7744] [Q&A] [answered]
According to `draft-ietf-dnssd-srp-06` [2.2.5.5. ](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-06.html#section-2.2.5.5)[Removing published services](https://www.ietf.org/archive/id/draft-ietf-dnssd-srp-06.html#name-removing-published-services), removal of service registration is done by a client using a lease option of zero, or the client removes all associated services when removing a host.

This matches with the available CLI that we have (notice the `remove` and `clear` options in [README_SRP_CLIENT](https://github.com/openthread/openthread/blob/main/src/cli/README_SRP_CLIENT.md), but no such commands for [README_SRP_SERVER](https://github.com/openthread/openthread/blob/main/src/cli/README_SRP_SERVER.md))

Does this mean that an SRP server can never remove published service entries by itself?  Are there any use cases where this would be desired?

**Comment by wgtdkp** [comment, marked as answer]
Yeah, there are currently no APIs for removing a registered service on the SRP server side. If there are use cases, we may consider adding it or contribution is welcome!

------

If you are looking for a API or CLI command to clear any registered services for testing or development, you can try `srp server disable` and `srp server enable` to re-enable the SRP server.

**Reply by suveshpratapa** [reply]
Thank you for clearing that @wgtdkp !
